Il2CppInspector.Common/Outputs/UnityHeaders/UnityHeader.cs

```diff
@@ -0,0 +1,81 @@
+            MetadataVersion = double.Parse(bits[0]);
+            MinVersion = new UnityVersion(bits[1]);
+            if (bits.Length == 2)
+                MaxVersion = MinVersion;
+            else if (bits[2] != "")
+                MaxVersion = new UnityVersion(bits[2]);
+        }
+
+        public override string ToString() {
+            var res = $"{MinVersion}";
+            if (MaxVersion == null)
+                res += "+";
+            else if (MaxVersion != MinVersion)
+                res += $" - {MaxVersion}";
+            return res;
+        }
+
+        public bool Contains(UnityVersion version) {
+            return version.CompareTo(MinVersion) >= 0 && (MaxVersion == null || version.CompareTo(MaxVersion) <= 0);
+        }
+
+        public string GetHeaderText() {
+            string resourceName = typeof(UnityHeader).Namespace + "." + HeaderFilename;
+            Assembly assembly = Assembly.GetCallingAssembly();
+            using Stream stream = assembly.GetManifestResourceStream(resourceName);
+            if (stream == null) {
+                throw new FileNotFoundException(resourceName);
+            }
+            using StreamReader reader = new StreamReader(stream);
+            string result = reader.ReadToEnd();
+            return result;
+        }
+
+        public static IEnumerable<UnityHeader> GetAllHeaders() {
+            string prefix = typeof(UnityHeader).Namespace + ".";
+            Assembly assembly = Assembly.GetCallingAssembly();
+            return assembly.GetManifestResourceNames()
+                .Where(s => s.StartsWith(prefix) && s.EndsWith(".h"))
+                .Select(s => new UnityHeader(s.Substring(prefix.Length)));
+        }
+
+        public static UnityHeader GetHeaderForVersion(string version) => GetHeaderForVersion(new UnityVersion(version));
+        public static UnityHeader GetHeaderForVersion(UnityVersion version) {
+            return GetAllHeaders().Where(v => v.Contains(version)).First();
+        }
+
+        public static List<UnityHeader> GuessHeadersForModel(Reflection.Il2CppModel model) {
+            List<UnityHeader> result = new List<UnityHeader>();
+            foreach (var v in GetAllHeaders()) {
+                if (v.MetadataVersion != model.Package.BinaryImage.Version)
+                    continue;
+                if (v.MetadataVersion == 21) {
+                    /* Special version logic for metadata version 21 based on the Il2CppMetadataRegistration.fieldOffsets field */
+                    var headerFieldOffsetsArePointers = (v.MinVersion.CompareTo("5.3.7") >= 0 && v.MinVersion.CompareTo("5.4.0") != 0);
+                    var binaryFieldOffsetsArePointers = (model.Package.Binary.FieldOffsets == null);
+                    if (headerFieldOffsetsArePointers != binaryFieldOffsetsArePointers)
+                        continue;
+                }
+                result.Add(v);
+            }
+            return result;
+        }
+    }
+}
```

Comment:
Please could you add short description comments (in UnityHeader.cs, UnityVersion.cs, CppDeclarations.cs and Namespace.cs) to the top of major properties and methods similar to the style we have in TypeInfo.cs and elsewhere so that in 6 months I remember what all this stuff is for :) Also make sure you have a blank line between a property and method.

You can also add copyright messages to all of the files you've edited across the various PRs if you wish (entirely optional but I like major contributors to be credited), for example:

```suggestion
/*
    Copyright 2017-2020 Katy Coe - http://www.hearthcode.org - http://www.djkaty.com
    Copyright 2020 Robert Xiao - <your desired URLs and/or emails>

    All rights reserved.
*/
```

Reply:
Done.